etc/init.d/zfs.in

```diff
@@ -0,0 +1,74 @@
+# ZoL userland configuration.
+
+# Run `zfs mount -a` during system start?
+ZFS_MOUNT='yes'
+
+# Run `zfs unmount -a` during system stop?
+ZFS_UNMOUNT='yes'
+
+# Run `zfs share -a` during system start?
+# nb: The shareiscsi, sharenfs, and sharesmb dataset properties.
+ZFS_SHARE='yes'
```

Comment:
> when backing up datasets to another server. I want to keep those share properties on the dataset (for easy recovery), but I don't want the backup server to be sharing those shares.

Fair enough, that's a valid point.

> Whether or not sharing should be on by default is it's own discussion.

However, this is not. "By default" means just that. Any local opinion you have besides that IS your decision. But "by default" (as in "useful/correct for the large majority"), it should be enabled.

Now, I get (and agree with you) that for YOU (and probably quite a number of users) automatic sharing should be off. But so should probably mounting� I need to think of the large majority, and that is those that do NOT send their data to another system and therefor needs/wants automatic mounting and sharing.

Now, since this is a local configuration file which is NOT automatically updated when/if it already exists, you should be safe from upgrades (once you changed it ONCE).